h/services/annotation_delete.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from datetime import datetime
+
+from h.events import AnnotationEvent
+from h.models import Annotation
+
+
+class AnnotationDeleteService(object):
+    def __init__(self, request):
+        self.request = request
+
+    def delete(self, annotation):
+        """
+        Deletes the given annotation.
+
+        :param annotation: the annotation to be deleted
+        :type annotation: h.models.Annotation
+        """
+        annotation = self.request.db.query(Annotation).get(annotation.id)
```

Comment:
What's the reason for the DB query here? Isn't `annotation` already the annotation model object?

Reply:
Yeah good point. Not needed. Just some leftovers from the conversion that aren't needed anymore.